// myItemStore/index.js
/*global console: false, require: false, module: false, exports: false */
var items = {
    0: {
        title: "Buy cats",
        description: "Buy more cats!",
        id: 0
    },
    1: {
        title: "Don't buy dogs",
        description: "Dogs are not for front end developers, don't buy",
        id: 1
    }
},
    lastId = 1;

function getItem(id, callback) {
    'use strict';
    var item = items[id];
    if (undefined !== item) {
        callback(200, item);
    } else {
        callback(404);
    }
}

function getAllItems(callback) {
    'use strict';
    callback(200, items);
}

function addItem(item, callback) {
    'use strict';
    var status;
    console.log(item);
    if (undefined === items[item.id]) {
        lastId += 1;
        while (undefined !== items[lastId]) {
            lastId += 1;
        }
        if (undefined === item.id) {
            item.id = lastId;
        }
        items[item.id] = item;
        status = 201;
    } else {
        status = 409;
    }
    callback(status, item.id);
}

function updateItem(id, item, callback) {
    'use strict';
    if (undefined !== items[id]) {
        console.log("HERE!");
        console.log(item);
        console.log(items[id]);
        items[id] = item;
        callback(200, id);
    } else {
        callback(404);
    }
}

function deleteItem(id, callback) {
    'use strict';
    var status;
    if (undefined !== items[id]) {
        delete items[id];
        status = 200;
    } else {
        status = 404;
    }
    callback(status);
}

module.exports = {
    getItem: getItem,
    getAllItems: getAllItems,
    addItem: addItem,
    updateItem: updateItem,
    deleteItem: deleteItem
};
